addons/mass_mailing/wizard/mail_compose_message.py

```diff
@@ -40,11 +40,29 @@ def get_mail_values(self, res_ids):
                         'mailing_model': self.model,
                         'mailing_domain': self.active_domain,
                 })
+
+            # Preprocess res.partners to batch-fetch from db
+            # if recipient_ids is present, it means they are partners 
+            # (the only object to fill get_default_recipient this way)
+            recipient_partners_ids = []
+            read_partners = {}
+            for res_id in res_ids:
+                mail_values = res[res_id]
+                if mail_values.get('recipient_ids'):
+                    # recipient_ids is a list of tuples at this point
+                    recipient_partners_ids.append(mail_values.get('recipient_ids')[0][1])
```

Comment:
Yes, as we target a stable version I told @kebeclibre  to avoid changing the generated values of other methods. I know the code is quite ugly but the purpose here is to make it work, not to beautify it.